### PR TITLE
Decouple get link logic from v2

### PR DESCRIPTION
**Pull request by yufeih**
This PR picks up the latest markdown extension changes.

#4078

### CODE REVIEW

File: docs/specs/xref.yml [hunk 1]
```diff
@@ -436,7 +436,7 @@ inputs:
 outputs:
   docs/a.json:
   docs/b.json: |
-    { "content": "<p>Link to <a href=\"xref:c#bookmark\">title</a></p>" }
+    { "content": "<p>Link to <a href=\"\">title</a></p>" }
```
Comment (live1206) [inline docs/specs/xref.yml]: is this a behavior change?

Comment (yufeih) [reply docs/specs/xref.yml]: it is, does it currently impact our comparison with v2?

Comment (live1206) [reply docs/specs/xref.yml]: I think it should be as it is written while uid not resolved

Comment (live1206) [reply docs/specs/xref.yml]: will this affect v2 as well?

Comment (yufeih) [reply docs/specs/xref.yml]: nop, it'll not affect v2.

Comment (live1206) [reply docs/specs/xref.yml]: then we are having a behavior change compared to v2

**Pull request merged**